src/Tidy/Codegen/Precedence.purs

```diff
@@ -181,7 +181,7 @@ precBinder2 a = case a of
   BinderOp _ _ -> binderParens a
   BinderInt (Just _) _ _ -> binderParens a
   BinderNumber (Just _) _ _ -> binderParens a
-  BinderConstructor _ _ -> binderParens a
+  BinderConstructor _ args | not $ Array.null args -> binderParens a
```

Comment:
I found that `precBinder2` is only used for the `binderCtor`. So, rather than defining a new `precBinder3`, I just updated `precBinder2` here.